### PR TITLE
First draft of event callback integration

### LINKED ISSUE #45: Log events
It would be nice to be able to log certain events, like connection start and connection end, to tools like http://grafana.org/.

To keep things generic, the user could specify a script on the commandline that is invoked as a "callback" with the event data (e.g. `{"name": "connect"}` or `{"name": "disconnect"}`) to stdin whenever an event happens.

@lgrahl probably wouldn't be a lot of effort, right?

### PR DESCRIPTION
**Pull request by dbrgn**
Should fix #45.

@lgrahl if this looks OK I can try to log more events.

### CODE REVIEW

File: saltyrtc/server/events.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+)
+from typing import List
+
+__all__ = ('Event',)
+
+
+@unique
+class Event(Enum):
+    INITIATOR_CONNECTED = 'initiator-connected'
+    RESPONDER_CONNECTED = 'responder-connected'
```
Comment (dbrgn) [inline saltyrtc/server/events.py]: Not sure if we should keep this as string values, or just use `auto()`.

Comment (lgrahl) [inline saltyrtc/server/events.py]: Either should be fine, both will be converted to hash values for lookup in the dict anyway, right?

Comment (dbrgn) [inline saltyrtc/server/events.py]: I think so, yes.

File: saltyrtc/server/events.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import collections
+from enum import (
```
Comment (lgrahl) [inline saltyrtc/server/events.py]: Please avoid `from` imports for external packages (unless nesting becomes too uncomfortable). I use this coding convention to be able to differentiate external and internal classes/functions easily in code without having to look at the imports.

File: saltyrtc/server/server.py [hunk 3]
```diff
@@ -45,7 +58,10 @@
 
 
 @asyncio.coroutine
-def serve(ssl_context, key, paths=None, host=None, port=8765, loop=None):
+def serve(
+        ssl_context, key, paths=None, host=None, port=8765, loop=None,
+        event_callbacks: Dict[Event, List[Coroutine]] = None
```
Comment (lgrahl) [inline saltyrtc/server/server.py]: No space before and after `=` (unless this changed in PEP8 for type hints? But that would be weird).

Comment (dbrgn) [inline saltyrtc/server/server.py]: This is in line with [PEP 484 recommendations](https://www.python.org/dev/peps/pep-0484/#arbitrary-argument-lists-and-default-argument-values). PEP8 refers to PEP484 for syntax in this case.

The thing is that `event_callbacks: Dict[Event, List[Coroutine]]=None` looks weird, as if `=None` were part of the type.

File: saltyrtc/server/events.py (hunk 2 above)
Comment (dbrgn) [inline saltyrtc/server/events.py]: `enum` is not an external package.

Comment (lgrahl) [inline saltyrtc/server/events.py]: I meant *external* regarding our code.

Comment (dbrgn) [inline saltyrtc/server/events.py]: Ah. Alright, I'll change that. I won't do it for `typing` values though, as in that case the signatures would get really long.

File: saltyrtc/server/server.py (hunk 3 above)
Comment (lgrahl) [inline saltyrtc/server/server.py]: They also do it like you in [PEP 484](https://www.python.org/dev/peps/pep-0484/#id48) so fair enough.

File: saltyrtc/server/__init__.py [hunk 4]
```diff
@@ -15,6 +15,8 @@
 from .server import *  # noqa
 # noinspection PyUnresolvedReferences
 from .util import *  # noqa
+# noinspection PyUnresolvedReferences
+from .events import *  # noqa
 
 __all__ = tuple(itertools.chain(
     ('bin',),
```
Comment (lgrahl) [inline saltyrtc/server/__init__.py]: Add events.__all__ here as well.

**Comment by dbrgn** [comment]
@lgrahl disconnected event added. ok now?

File: saltyrtc/server/server.py [hunk 5]
```diff
@@ -228,11 +262,14 @@ def handle_client(self):
         tasks = [self.task_loop()]
 
         # Task: Poll for messages
+        path = binascii.hexlify(self.path.initiator_key).decode('ascii')
```
Comment (lgrahl) [inline saltyrtc/server/server.py]: Should probably be named `hex_path` as above to avoid confusion.

Comment (dbrgn) [inline saltyrtc/server/server.py]: Done

**Pull request merged**